Source/UI/Coordinators/ShowTermsCoordinator.swift

```diff
@@ -0,0 +1,102 @@
+class ShowTermsCoordinator: FlowCoordinator {
+    struct Input {
+        enum PresentationStyle {
+            case replaceNavigationFlow
+            case pushToExistingNavigationFlow
+
+            fileprivate var hasBackButton: Bool {
+                switch self {
+                case .replaceNavigationFlow:
+                    return false
+                case .pushToExistingNavigationFlow:
+                    return true
+                }
+            }
+        }
+
+        let terms: Terms
+        let loginFlowVariant: LoginMethod.FlowVariant
+        let presentationStyle: PresentationStyle
+    }
+
+    enum Output {
+        case success
+        case cancel
+        case back
+    }
+
+    let navigationController: UINavigationController
+    let configuration: IdentityUIConfiguration
+
+    var child: ChildFlowCoordinator?
+
+    init(navigationController: UINavigationController, configuration: IdentityUIConfiguration) {
+        self.navigationController = navigationController
+        self.configuration = configuration
+    }
+
+    func start(input: Input, completion: @escaping (Output) -> Void) {
+        self.showAcceptTermsView(
+            for: input.terms,
+            loginFlowVariant: input.loginFlowVariant,
+            presentationStyle: input.presentationStyle,
+            completion: completion
+        )
+    }
+}
+
+extension ShowTermsCoordinator {
+    private func showAcceptTermsView(
+        for terms: Terms,
+        loginFlowVariant: LoginMethod.FlowVariant,
+        presentationStyle: Input.PresentationStyle,
+        completion: @escaping (Output) -> Void
+    ) {
+        let navigationSettings = NavigationSettings(
+            cancel: { completion(.cancel) },
+            back: presentationStyle.hasBackButton ? { completion(.back) } : nil
+        )
+        let viewModel = TermsViewModel(
+            terms: terms,
+            loginFlowVariant: loginFlowVariant,
+            appName: self.configuration.appName,
+            localizationBundle: self.configuration.localizationBundle
+        )
+
+        let viewController = TermsViewController(configuration: self.configuration, navigationSettings: navigationSettings, viewModel: viewModel)
+        viewController.didRequestAction = { [weak self] action in
+            switch action {
+            case .acceptTerms:
+                completion(.success)
+            case let .learnMore(summary):
+                self?.showTermsSummaryView(summary)
+            case let .open(url):
+                self?.present(url: url)
+            case .back:
+                completion(.back)
+            case .cancel:
+                completion(.cancel)
+            }
+        }
+
+        switch presentationStyle {
+        case .pushToExistingNavigationFlow:
+            self.navigationController.pushViewController(viewController, animated: true)
+        case .replaceNavigationFlow:
+            self.navigationController.viewControllers = [viewController]
```

Comment:
Do we need the presentation style? Is it enough to base this on whether `self.navigationController.viewControllers.count > 0` ? If yes then we continue a flow, else we're starting a new flow.

Reply:
Existence of back button is also then implicit.

Reply:
Or now I'm thinking maybe there's something with a loginFlowVariant we're missing. Because there's no variant on the "static present updated terms flow" ?

Reply:
Yes that check should work, I'll change it 👌

True, the `loginFlowVariant` doesn't really apply when presenting the updated terms, for now it falls back to the login variant in that case.